sklearn/inspection/_plot/tests/test_boundary_decision_display.py

```diff
@@ -577,16 +575,85 @@ def test_class_of_interest_multiclass(pyplot, response_method):
             class_of_interest=class_of_interest_idx,
         )
 
-    # TODO: remove this test when we handle multiclass with class_of_interest=None
-    # by showing the max of the decision function or the max of the predicted
-    # probabilities.
-    err_msg = "Multiclass classifiers are only supported"
-    with pytest.raises(ValueError, match=err_msg):
+
+@pytest.mark.parametrize("response_method", ["predict_proba", "decision_function"])
+def test_multiclass_plot_max_class(pyplot, response_method):
+    """Check plot correct when plotting max multiclass class."""
+    X, y = load_iris_2d_scaled()
+    clf = LogisticRegression().fit(X, y)
+
+    disp = DecisionBoundaryDisplay.from_estimator(
+        clf,
+        X,
+        response_method=response_method,
+    )
+
+    grid = np.concatenate([disp.xx0.reshape(-1, 1), disp.xx1.reshape(-1, 1)], axis=1)
+    response = getattr(clf, response_method)(grid)
+    assert_allclose(response.reshape(*disp.response.shape), disp.response)
+
+
+@pytest.mark.parametrize(
+    "multiclass_colors",
+    [
+        "plasma",
+        ["red", "green", "blue"],
+    ],
+)
+@pytest.mark.parametrize("plot_method", ["contourf", "contour", "pcolormesh"])
+def test_multiclass_colors_cmap(pyplot, plot_method, multiclass_colors):
+    """Check correct cmap used  for all `multiclass_colors` inputs."""
+    import matplotlib as mpl
+
+    X, y = load_iris_2d_scaled()
+    clf = LogisticRegression().fit(X, y)
+
+    disp = DecisionBoundaryDisplay.from_estimator(
+        clf,
+        X,
+        plot_method=plot_method,
+        multiclass_colors=multiclass_colors,
+    )
+
+    cmaps = []
+    if multiclass_colors == "plasma":
+        cmap = mpl.pyplot.get_cmap(multiclass_colors, len(clf.classes_))
+        for class_idx, primary_color in enumerate(cmap.colors):
+            r, g, b, _ = primary_color
+            class_cmap = mpl.colors.LinearSegmentedColormap.from_list(
+                f"colormap_{class_idx}", [(1.0, 1.0, 1.0, 1.0), (r, g, b, 1.0)]
+            )
+            cmaps.append(class_cmap)
+    else:
+        for class_idx, class_color in enumerate(multiclass_colors):
+            r, g, b, _ = mpl.colors.to_rgba(class_color)
+            class_cmap = mpl.colors.LinearSegmentedColormap.from_list(
+                f"colormap_{class_idx}", [(1.0, 1.0, 1.0, 1.0), (r, g, b, 1.0)]
+            )
+            cmaps.append(class_cmap)
+
+    for idx, quad in enumerate(disp.surface_):
+        qq = quad.cmap._segmentdata
+        testcc = cmaps[idx]._segmentdata
+        assert (
+            quad.cmap == cmaps[idx]
```

Comment:
If this is failing, we can probably check some attributes such as the name, and the limit (`.get_under()` and `_get_over()`). I'll check if there is any other meaningful attribute.

Reply:
The attribute `N` might also be useful. Basically, with the lower bound higher bound and the level of quantization, we are almost certain that we will get the same colormap.

Reply:
So I think this is failing because in mpl min version (3.3.4), `__eq__` is not implemented for `LinearSegmentedColormap`. Trying to find the PR or doc for this, so I can make a good comment in the code but not successful.

I think I will just skip for older mpl version, WDYT?

Reply:
Ah okay this was added in 3.5 in `colormaps` class, so it will check for equality even when its not the same object, see: https://github.com/matplotlib/matplotlib/pull/20227

Reply:
Yep that's why I thought that we could check the couple of public of attributes allowing to check that everything is as expected: name and bounds

Reply:
Ah yup, I can just manually test what `__eq__` checks. Will amend!

Reply:
I've just checked that `_lut` is equal, which seems to be the main think that is checked in the new `__eq__` method of `Colormaps` in mpl. Happy to change/add.

Reply:
Not 100% that using the private `_lut` is the best option (note need to init first). I do think this is the best item to compare (as it is what is done in `__eq__` method, and e.g., `get_under`() etc do use `._lut`, e.g.:

```python
    def get_under(self):
        """Get the color for low out-of-range values."""
        if not self._isinit:
            self._init()
        return np.array(self._lut[self._i_under])
```

Happy to change.